target/linux/realtek/files-5.10/drivers/net/ethernet/rtl838x_eth.c

```diff
@@ -1263,6 +1264,7 @@ static int rtl838x_hw_receive(struct net_device *dev, int r, int budget)
 
 	pr_debug("---------------------------------------------------------- RX - %d\n", r);
 	spin_lock_irqsave(&priv->lock, flags);
+	INIT_LIST_HEAD(&rx_list);
```

Comment:
This could probably be replaced with `LIST_HEAD(rx_list)`, that would avoid the need to declare `struct list_head rx_list`

Reply:
I don't see this pattern upstream for some reason. All usages of netif_receive_skb_list use INIT_LIST_HEAD

Reply:
Then again I also see https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?h=next-20221028&id=d32f211adb6aa179c00ee1c251315d1ef1433a38 . Hrm,

Reply:
Do note INIT_LIST_HEAD is called after a spinlock. Not sure what the consequence of doing that outside is.

Reply:
Fixed anyway. It's done elsewhere besides ethernet drivers. Looks like someone needs to run coccinelle upstream.